openpgp/query.go

```diff
@@ -0,0 +1,134 @@
+/*
+   Hockeypuck - OpenPGP key server
+   Copyright (C) 2012-2014  Casey Marshall
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published by
+   the Free Software Foundation, version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package openpgp
+
+import (
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type query interface {
+	UpdatePubkey(e sqlx.Execer, p *Pubkey) (sql.Result, error)
```

Comment:
I'd remove the `sqlx.Execer` argument from these methods, and leave it to the implementation of this interface to have the necessary context (connection & transaction handles, etc.) to carry out the operation.